ArduPlane/Plane.h

```diff
@@ -1113,6 +1126,10 @@ class Plane : public AP_Vehicle {
     float get_throttle_input(bool no_deadzone=false) const;
     float get_adjusted_throttle_input(bool no_deadzone=false) const;
 
+    // support for NAV_SCRIPTING mission command
+    bool nav_scripting_active(void) override;
+    bool nav_scripting_update(bool completed, float throttle_pct, float roll_rate_dps, float pitch_rate_dps, float yaw_rate_dps) override;
```

Comment:
It depends a bit upon the vehicle but in general a lua script can always control the vehicle in Guided mode and a script can also change the vehicle's flight mode so a script could change the vehicle to Guided mode and control it in a variety of different ways.

This PR is adding support for Lua scripts to control the vehicle (a Plane) not only while in Guided mode but also within Auto mode.

Reply:
I mean as it stands, doesn't the function take essentially a mission item as an argument? How would you call this from a regular timer-style function?

Reply:
@arikrupnik, by "regular timer-style function" I guess you mean a C++ function/method within the regular vehicle code (not a script)?  Some custom method that a developer has written that runs at a regular interval because it is within the Scheduler array?  If I understand the question correctly then I don't think that this method could or should make use of this new code in this PR.

Reply:
@rmackay9 I think I expressed myself poorly. I imagine using this new API similarly to `set_target_angle_and_climbrate()`, from anywhere in a LUA script. It appears that @tridge's intent is to allow its use only in response to a mission item. There must be a reason not to generalize its use to allow it from anywhere. I'd like to understand it. The intent is explicit in several places, including `commands_logic.cpp`:
```c
    if (!nav_scripting_active()) {
        return false;
    }
```